lib/view_model/exchange/exchange_trade_view_model.dart

```diff
@@ -115,16 +109,15 @@ abstract class ExchangeTradeViewModelBase with Store {
 
   @action
   Future<void> _updateTrade() async {
+    if (_provider == null) throw ExchangeProviderUnsupportedException(trade.provider);
+
```

Comment:
you haven't added `TradeProviderUnsupportedItem` to the items in this class.

also, there is no need for this line anymore